### ISSUE TITLE
Spread

### ISSUE THREAD

**Issue by matthewrobb** [matthewrobb/six#10]
http://wiki.ecmascript.org/doku.php?id=harmony:spread

**Comment by insidewhy** [comment]
Hey my spread support change has landed on the harmony branch!

Could you release a new version of esprima-six that includes it? Then I can prepare a pull request that uses it and adds spread support.

Thanks! Don't rush I know you're busy :) I'll push the package esprima-six-jpike while I'm waiting to test.

**Comment by matthewrobb** [comment]
Done. Published to NPM.

**Comment by insidewhy** [comment]
That message appeared whilst I was looking at it... Cool thanks!

**Comment by matthewrobb** [comment]
Not a problem. I am going to be going ahead of my family to the Seattle area to start a new job and find us a home. In my off time I imagine I will be putting in a lot of time to this project. I am already gearing for that. We should chat some time about road map etc.

**Comment by insidewhy** [comment]
Cool, I'm working on a parser at the moment using my own parsing system based on PEGs. It allows creating a parser that constructs an AST from a single grammar.

I didn't wanna write it in pure JavaScript though that's why I thought using six would be more fun. I just thought it would be fun to add in all the bits I wanted to use that weren't already present.

**Comment by matthewrobb** [comment]
Awesome! Your contributions are definitely appreciated. My plan is to get some of the key features roughly implemented and then switch to focusing on the website ( sixlang.org ), documentation, and general community outreach.

**Comment by insidewhy** [comment]
Oh I wonder, would you mind if I switched the test system to being in six rather than js?

Since my change to add support for ".six" file extension you can do it with:

mocha --compilers six:six

**Comment by matthewrobb** [comment]
Yeah sounds great. I didn't put any time in to developing a long term testing solution and I'd definitely like to get to that point. Ideally the test suite would be a generic enough suite that it's able to validate functional harmony code.

**Comment by insidewhy** [comment]
I think I meant rest parameters rather than spreads in my first comment. Oops. Anyway, spread support is now on the esprima harmony branch!

**Comment by insidewhy** [comment]
Doah, no it isn't. Just fixed spread parameters for arrow functions. Sorry I'm really high.

**Comment by matthewrobb** [comment]
I think you mean Rest parameters on this one. Spread is an operator not normally used in parameters unless used in destructuring parameters.

``` JavaScript
// Rest
function(...rest){}

// Spread
var arr = [1,2,3, ...rest]
```
